src/shared/components/palette/datetime-picker/datetime-picker.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, ElementRef, Inject, Input, OnInit, ViewChild, ViewEncapsulation } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Moment } from 'moment/moment';
+import {
+  NGX_MAT_DATE_FORMATS,
+  NgxMatDateAdapter,
+  NgxMatDateFormats,
+  NgxMatDatetimePicker
+} from '@angular-material-components/datetime-picker';
+import { NgxMatMomentAdapter, NGX_MAT_MOMENT_DATE_ADAPTER_OPTIONS } from '@angular-material-components/moment-adapter';
+import { ThemePalette } from '@angular/material';
+
+import { AbstractFormFieldComponent } from '../base-field/abstract-form-field.component';
+import { CaseField } from '../../../domain';
+import { CUSTOM_MOMENT_FORMATS } from './datetime-picker-utils';
+import { FormatTranslatorService } from '../../../services/case-fields/format-translator.service';
+
+@Component({
+  selector: 'ccd-datetime-picker',
+  templateUrl: './datetime-picker.component.html',
+  styleUrls: ['./datetime-picker.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  providers: [
+    {provide: NGX_MAT_DATE_FORMATS, useValue: CUSTOM_MOMENT_FORMATS},
+    {provide: NgxMatDateAdapter, useClass: NgxMatMomentAdapter},
+    {provide: NGX_MAT_MOMENT_DATE_ADAPTER_OPTIONS, useValue: {useUtc: true}}]
+})
+
+export class DatetimePickerComponent extends AbstractFormFieldComponent implements OnInit {
+
+  public showSpinners = true;
+  public showSeconds = false;
+  public touchUi = false;
+  public enableMeridian = false;
+  public stepHour = 1;
+  public stepMinute = 1;
+  public stepSecond = 1;
+  public color: ThemePalette = 'primary';
+  public disableMinute = true;
+  public hideTime = true;
+  public hideMinutes = true;
+  public startView = 'month';
+  public yearSelection = false;
+  public checkTime = true;
+  public stringEdited = false;
+  @Input() public dateControl: FormControl = new FormControl(new Date());
+
+  @ViewChild('picker') datetimePicker: NgxMatDatetimePicker<any>;
+  @ViewChild('input') inputElement: ElementRef<HTMLInputElement>;
+  private dateTimeEntryFormat;
+
+  constructor(private readonly formatTranslationService: FormatTranslatorService,
+              @Inject(NGX_MAT_DATE_FORMATS) private ngxMatDateFormats: NgxMatDateFormats) {
+
+    super();
+  }
+
+  ngOnInit(): void {
```

Comment:
_Nitpick:_
We agreed at the last Tech Forum that these will always be `public ngOnInit(): void` going forward.

Reply:
Done